src/Init/SimpLemmas.lean

```diff
@@ -287,8 +287,8 @@ theorem not_decide_eq_true [h : Decidable p] : ((!decide p) = true) = ¬ p := by
 @[simp] theorem cond_true (a b : α) : cond true a b = a := rfl
 @[simp] theorem cond_false (a b : α) : cond false a b = b := rfl
 
-@[simp] theorem beq_self_eq_true [BEq α] [LawfulBEq α] (a : α) : (a == a) = true := LawfulBEq.rfl
-theorem beq_self_eq_true' [DecidableEq α] (a : α) : (a == a) = true := by simp
+theorem beq_self_eq_true [BEq α] [ReflBEq α] (a : α) : (a == a) = true := BEq.rfl
```

Comment:
Why did you decide to remove this `simp` annotation?

Reply:
Ah, right. I guess in this case you could even deprecate the two variants of `beq_self_eq_true` in favor of `BEq.refl`.